python/test/test_typing.py

```diff
@@ -14,7 +14,39 @@ def test_numpy_function():
     print(RuntimeType.infer(np.array(["A", "B"])))
 
 
-def test_typing_hint():
+def test_typing_infer():
+    # Is there an argument for wrapping these in RuntimeType instances,
+    # and work towards simplifying the python type annotations?
+    assert RuntimeType.infer(23) is 'i32' # noqa: F632
+    assert RuntimeType.infer(12.) is 'f64' # noqa: F632
+    assert RuntimeType.infer('hello') is 'String' # noqa: F632
+    assert RuntimeType.infer(lambda: True) is 'CallbackFn' # noqa: F632
+    
+    assert RuntimeType.infer(object(), py_object=True) is 'ExtrinsicObject' # noqa: F632
+    # TODO: This one seems strange: Why not have the usual error if types don't match?
+    assert RuntimeType.infer([1, True], py_object=True) == 'Vec<ExtrinsicObject>'
+
+    # Note that here we assert "==" rather than "is":
+    infer_vec = RuntimeType.infer(["A", "B"])
+    assert infer_vec == 'Vec<String>'
+    assert isinstance(infer_vec, RuntimeType)
+
+    infer_tuple = RuntimeType.infer((12., True, "A"))
+    assert infer_tuple == '(f64, bool, String)'
+    assert isinstance(infer_tuple, RuntimeType)
+
+    with pytest.raises(UnknownTypeException, match=re.escape("<class 'object'>")):
+        RuntimeType.infer(object())
+    
+    # TODO: Raising an exception in __repr__ seems strange: Why not raise exception earlier, like infer(object())?
```

Comment:
It's mainly because of `UnknownType`. If you wanted to check if a Python value was similar to a Rust type, and the value was an empty list, then throwing an error upon inferring the type would be wrong-- an empty list can still be similar to a `Vec<T>` for any `T`. It might not be necessary now that `assert_is_similar` is gone (these checks are now handled incrementally during the data load process).

Reply:
I'll try a separate small PR to see if this can be simplified, as you suggest.